_posts/2019-07-11-deploy-spring-boot-kubernetes.md

```diff
@@ -0,0 +1,139 @@
+The second line tells the image that it should work from the /tmp directory.
+The third line copies the compiled JAR (which is compiled with the `mvn clean install` command) file from your target folder to the Docker image (you might have to rename the file depending on the name of your project).
+
+Finally, we tell our image to use the java command as entry point, meaning that once the Docker image starts running, it has to execute that command.
+
+You can now push this image to your favorite Docker registry, as Kubernetes will need to pull this image from the registry later. If you do not have a Docker registry, I suggest using Docker Hub. If you are using Docker Hub, you can use the following commands to build and push your application to the registry:
+
+```
+docker build -t DOCKER_HUB_USERNAME/APPLICATION_NAME .
+docker push DOCKER_HUB_USERNAME/APPLICATION_NAME
+````
+
+## Where the magic happens: deploying your application on the Kubernetes cluster
+
+All Kubernetes configurations are written in YAML. The reason for this is that Kubernetes configuration files are meant to be easily readable by the human eye and the Kubernetes team decided to use YAML instead of JSON.
+
+### Deployment
+
+Now that the Docker image is created we can now deploy it on the Kubernetes cluster. 
+First we need to create a deployment configuration file. This file contains the configuration on how the application should run.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kubernetesdemo
+  labels:
+    app: kubernetesdemo
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kubernetesdemo
+    spec:
+      containers:
+        - name: kubernetesdemo
+          image: yolanv/kubernetesdemo
+          ports:
+            - containerPort: 8080
+```
+There's a lot going on here which I will be explaining step by step.
+TThe first 2 lines are telling which Kubernetes API version is being used and what kind of Kubernetes [object](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/){:target="_blank" rel="noopener noreferrer"} is being applied. As we want to create a new Deployment, we use the Deployment object (easy, right?).
+
+Lines 3 to 6 are just basic metadata tags so the developer knows which application (s)he is working with. This does not affect the behavior of the application in any way.
+
+Lines 7 to 21 are specifying how the container should be made and which image it has to run. This is the image that we created with the Dockerfile earlier in this post. After that, it describes the port that the container should listen to, which is 8080 in this case. The replica value specifies how many 'instances' (also called [Pods](https://kubernetes.io/docs/concepts/workloads/pods/pod/){:target="_blank" rel="noopener noreferrer"}) that should be running. If the application is expecting a lot of requests, it might be useful to declare a higher number of replicas instead of 1.
```

Comment:
```suggestion
Lines 7 to 21 are specifying how the container should be made and which image it has to run. This is the image that we created with the Dockerfile earlier in this post. After that, it describes the port that the container should listen to, which is 8080 in this case. The replica value specifies how many 'instances' (also called [Pods](https://kubernetes.io/docs/concepts/workloads/pods/pod/){:target="_blank" rel="noopener noreferrer"}) that should be running. If the application is expecting a lot of requests, it might be useful to declare a higher number of replicas instead of only one.
```